docs/migrating/incremental-adoption.md

```diff
@@ -78,7 +76,7 @@ module.exports = {
 
 To learn more about rewrites, take a look at our [documentation](/docs/api-reference/next.config.js/rewrites.md).
 
-> This feature was introduced in [Next.js 9.5](https://nextjs.org/blog/next-9-5) and up. If you’re using older versions of Next.js, please upgrade before trying it out.
+> **Note:** If you are incrementally migrating to a dynamic route (e.g. `[slug].js`) and using `fallback` for your `rewrites`, ensure you consider the case where pages are not found. By default, Next.js will match the dynamic route and end the routing phase. Using `notFound: true` in `getStaticProps` would return the `404.js` page content, or the default 404 page. If this is not desired, you can use `getServerSideProps` with `stale-while-revalidate` Cache-Control headers instead of `notFound: true`. Then, you can _manually_ proxy your existing backend using something like [http-proxy](https://github.com/vercel/next.js/discussions/38839#discussioncomment-3744442).
```

Comment:
```suggestion
> **Note:** If you are incrementally migrating to a dynamic route (e.g. `[slug].js`) and using `fallback: true` or `fallback: 'blocking'` along with a fallback `rewrite`, ensure you consider the case where pages are not found. When Next.js matches the dynamic route it stops checking any further routes. Using `notFound: true` in `getStaticProps` will return the 404 page without applying the fallback `rewrite`. If this is not desired, you can use `getServerSideProps` with `stale-while-revalidate` Cache-Control headers when returning your props. Then, you can _manually_ proxy to your existing backend using something like [http-proxy](https://github.com/vercel/next.js/discussions/38839#discussioncomment-3744442) instead of returning `notFound: true`.
```